python/jittor/test/test_mpi.py

```diff
@@ -10,24 +10,50 @@
 import os, sys
 import jittor as jt
 import numpy as np
+mpi = jt.compile_extern.mpi
 
-def main():
-    print("test mpi_test")
-    assert jt.compile_extern.mpi_ops.mpi_test("").data == 123
-    if jt.compile_extern.nccl_ops:
-        print("test test_with_mpi")
-        with jt.flag_scope(use_cuda=1):
-            assert jt.compile_extern.nccl_ops.nccl_test("test_with_mpi").data == 123
-
-@unittest.skipIf(jt.compile_extern.has_mpi is None, "no mpi found")
+@unittest.skipIf(mpi is None, "no inside mpirun")
 class TestMpi(unittest.TestCase):
-    def test(self):
+    def test_mpi_test_op(self):
+        assert jt.compile_extern.mpi_ops.mpi_test("").data == 123
+
+    @unittest.skipIf(jt.compile_extern.nccl_ops is None, "no inccl")
+    @jt.flag_scope(use_cuda=1)
+    def test_nccl_with_mpi(self):
+        assert jt.compile_extern.nccl_ops.nccl_test("test_with_mpi").data == 123
+
+    def test_mpi_broadcast(self):
+        a = np.zeros(100) + mpi.world_rank()
+        mpi.broadcast(a, 0)
```

Comment:
Use all different threads for broadcast to test？

Reply:
yes